tests/test_sampler.py

```diff
@@ -131,13 +130,18 @@ def test_sampler_clear_iterations():
 
 @pytest.mark.parametrize("backend_name", ["SLOS", "CliffordClifford2017"])
 def test_sampler_iterator(backend_name):
-    c = BS() // PS(phi=pcvl.P("phi1")) // BS()
-    p = pcvl.Processor(backend_name, c)
+    phi = pcvl.P("phi1")
+    c = BS() // PS(phi=phi) // BS()
+    phi.set_value(0)
+    p = pcvl.Processor(backend_name, c, noise=NoiseModel(.5))
+    p.min_detected_photons_filter(2)
+    p.with_input(pcvl.BasicState([1, 1]))
     sampler = Sampler(p)
     iteration_list = [
         {"circuit_params": {"phi1": 0.5}, "input_state": pcvl.BasicState([1, 1]), "min_detected_photons": 1},
-        {"circuit_params": {"phi1": 0.9}, "input_state": pcvl.BasicState([1, 1]), "min_detected_photons": 1},
-        {"circuit_params": {"phi1": 1.57}, "input_state": pcvl.BasicState([1, 0]), "min_detected_photons": 1}
+        {"circuit_params": {"phi1": 0.9}, "input_state": pcvl.BasicState([1, 1]), "max_samples": 20},
+        {"circuit_params": {"phi1": 1.57}, "input_state": pcvl.BasicState([1, 0]), "min_detected_photons": 1, "max_shots": 30},
+        {}  # Test default parameters
```

Comment:
Please add `noise` in the test for the `add_iteration` too. It is the only one left behind :-(